samples/ScratchIslandApp/WindowExe/SampleBaseWindow.h

```diff
@@ -0,0 +1,228 @@
+        {
+        case WM_DPICHANGED:
+        {
+            return HandleDpiChange(_window.get(), wparam, lparam);
+        }
+
+        case WM_DESTROY:
+        {
+            PostQuitMessage(0);
+            return 0;
+        }
+
+        case WM_SIZE:
+        {
+            UINT width = LOWORD(lparam);
+            UINT height = HIWORD(lparam);
+
+            switch (wparam)
+            {
+            case SIZE_MAXIMIZED:
+                [[fallthrough]];
+            case SIZE_RESTORED:
+                if (_minimized)
+                {
+                    _minimized = false;
+                    OnRestore();
+                }
+
+                // We always need to fire the resize event, even when we're transitioning from minimized.
+                // We might be transitioning directly from minimized to maximized, and we'll need
+                // to trigger any size-related content changes.
+                OnResize(width, height);
+                break;
+            case SIZE_MINIMIZED:
+                if (!_minimized)
+                {
+                    _minimized = true;
+                    OnMinimize();
+                }
+                break;
+            default:
+                // do nothing.
+                break;
+            }
+            break;
+        }
+        case CM_UPDATE_TITLE:
+        {
+            SetWindowTextW(_window.get(), _title.c_str());
+            break;
+        }
+        }
+
+        return DefWindowProc(_window.get(), message, wparam, lparam);
+    }
+
+    // DPI Change handler. on WM_DPICHANGE resize the window
+    [[nodiscard]] LRESULT HandleDpiChange(const HWND hWnd, const WPARAM wParam, const LPARAM lParam)
+    {
+        _inDpiChange = true;
+        const HWND hWndStatic = GetWindow(hWnd, GW_CHILD);
+        if (hWndStatic != nullptr)
+        {
+            const UINT uDpi = HIWORD(wParam);
+
+            // Resize the window
+            auto lprcNewScale = reinterpret_cast<RECT*>(lParam);
+
+            SetWindowPos(hWnd, nullptr, lprcNewScale->left, lprcNewScale->top, lprcNewScale->right - lprcNewScale->left, lprcNewScale->bottom - lprcNewScale->top, SWP_NOZORDER | SWP_NOACTIVATE);
+
+            _currentDpi = uDpi;
+        }
+        _inDpiChange = false;
+        return 0;
+    }
+
+    virtual void OnResize(const UINT width, const UINT height) = 0;
+    virtual void OnMinimize() = 0;
+    virtual void OnRestore() = 0;
+
+    RECT GetWindowRect() const noexcept
+    {
+        RECT rc = { 0 };
+        ::GetWindowRect(_window.get(), &rc);
+        return rc;
+    }
+
+    HWND GetHandle() const noexcept
+    {
+        return _window.get();
+    }
+
+    float GetCurrentDpiScale() const noexcept
+    {
+        const auto dpi = ::GetDpiForWindow(_window.get());
+        const auto scale = static_cast<float>(dpi) / static_cast<float>(USER_DEFAULT_SCREEN_DPI);
+        return scale;
+    }
+
+    //// Gets the physical size of the client area of the HWND in _window

```

Comment:
Double commented here and below?

Reply:
Huh, it's double commented in `BaseWindow.h` too